packages/create-react-app/index.js

```diff
@@ -264,6 +270,30 @@ function checkAppName(appName) {
   }
 }
 
+function moveReactScriptsToDev() {
+  var packagePath = path.join(process.cwd(), 'package.json');
+  if (!pathExists(packagePath)) {
+    return;
+  }
+
+  var packageJson = require(packagePath);
+  if (typeof packageJson.dependencies === 'undefined') {
+    return;
+  }
+
+  var packageVersion = packageJson.dependencies['react-scripts']
```

Comment:
This also needs to work when the name of the package isn't `react-scripts`. A custom package can specified using the `--scripts-version` option, e.g.:
```
create-react-app my-app --scripts-version custom-react-scripts my-app
```
In this case the name of the script package can be something else than `react-scripts`, and we'll have to find it from the dependencies. Perhaps instead of looking for the key `react-scripts` we could find the entry with the value of `packageToInstall` as its value, though I'm not sure if that works in 100% of the cases.

Ideally we'd be able to install the package as a `devDependency` to begin with, so we wouldn't have to modify it afterwards, but it doesn't seem to be possible to install `dependencies` and `devDependencies` in a single command. Or we could write the (dev)dependencies to package.json when we initialize it earlier, but then we won't know which versions to put there... unless we run something like `yarn upgrade` but that doesn't exist for `npm` 😣 

Do you have any ideas how to fix it in a way that it still works with custom script packages?